test_expect_failure/missing_direct_deps/internal_deps/custom-jvm-rule.bzl

```diff
@@ -1,13 +1,8 @@
 def _custom_jvm_impl(ctx):
     print(ctx.label)
     transitive_compile_jars = _collect(ctx.attr.deps)
-    return struct(
-        providers = [
-            java_common.create_provider(
-                transitive_compile_time_jars = transitive_compile_jars,
-            ),
-        ],
-    )
+    providers = [JavaInfo(output_jar = jar, compile_jar = jar) for jar in transitive_compile_jars]
+    return [java_common.merge(providers)]
```

Comment:
doesn't this mean that the new merged JavaInfo will have `transitive_compile_jars` as top level dependencies? Currently they are transitive and not direct.

Reply:
That's true. I'm not sure how to create a provider that only has transitive jars and also I'm not sure we should. What is the purpose of `custom_jvm` rule? Why does it only have `deps`?